apis/keda/v1alpha1/scaledjob_types_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+)
+
+func TestScaledJob(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedMax int64
+		expectedMin int64
+		maxReplicas *int32
+		minReplicas *int32
+	}{
+		{
+			name:        "MaxReplicaCount is set to 10 and MinReplicaCount is set to 1",
+			expectedMax: 9,
```

Comment:
I think that you should put 10 here xD

Reply:
https://github.com/kedacore/keda/blob/11ec80b022d00954171eb8c8f8df420b774bfdc5/apis/keda/v1alpha1/scaledjob_types.go#L117

Reply:
is that correct? O.O MaxReplicaCount should be modified in that way?
If I set minReplicaCount=10 and maxReplicaCount=10, should we return 0? This sounds weird

Reply:
if minReplicaCount is greater than  minReplicaCount, will MaxReplicaCount() return a negative value?

Reply:
[It's a calculation for the specific case where minReplicas are set to 1 for scaledJobs](https://github.com/kedacore/keda/pull/3425#pullrequestreview-1055953003)

Reply:
> if minReplicaCount is greater than minReplicaCount, will MaxReplicaCount() return a negative value?

that's a good q. We should catch this in the controller, but we should probably deal with it here as well. What value do you propose in this case? `if min > max then max?`

Reply:
I'd set `if min > max then max` as temporal solution, but IMO we should review it in depth in the future